.gitignore

```diff
@@ -31,7 +31,7 @@ compile_commands.json
 .cache/
 
 # IDE/VSCode
-.vscode/
+#.vscode/
```

Comment:
What we want is to provide good defaults, but not run the risk of developers committing their private vs code configuration. Is there a way to do this with VS code?

Reply:
Yes but it would be beneficial to use the same vscode configuration.
Some ideas:
- commit only specific config files, like the ``.vscode/tasks.json`` , ``.vscode/launch.json``, ``.vscode/c_cpp_properties.json`` files
- commit files/folders with ``.default`` suffix, the developers could decide if they want to use the configurations or not

Reply:
Let's commit .vscode/ with the normal files, but add a git-hook in the pre-commit to ensure that changes aren't committed by accident.

Also, it should still be possible for users to add workspace changes, such as tasks or settings, without modifying the .vscode folder.